fix(scripts/Quest): The Smallest Creatures

grimgravy opened the pull request:
<!--- (**********************************)
      (** Fill in the following fields **)
      (**********************************) --->

**Changes proposed:**

-  fix: quest can be completed
-  fix the aura of invisibility works only for player
-  fix: when controlling marmot, you can cancel the control aura without a problem

**Issues addressed:**

Closes [#11332 ](https://github.com/TrinityCore/TrinityCore/issues/11332)


**Tests performed:**

tested in game

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] 
- [ ] 


<!--- Notes
- Enable the setting "[√] Allow edits from maintainers." when creating your pull request.
- If this PR only contains SQL files, open a new issue instead and post or link the SQL in the issue.
- When adding new SQL files, name them 9999_99_99_99_db_name.sql to reduce the chance of possible merge conflicts.
--->

meji46 commented on src/server/game/Spells/SpellMgr.cpp:
```diff
@@ -4948,6 +4948,13 @@ void SpellMgr::LoadSpellInfoCorrections()
+    ApplySpellFix({ 38544 }, [](SpellInfo* spellInfo)
+    {
+        spellInfo->_GetEffect(EFFECT_0).MiscValueB = 65;
+        spellInfo->_GetEffect(EFFECT_1).ApplyAuraName = SPELL_AURA_DUMMY;
```
This is a hack. Player should be invisible while controlling the marmot

grimgravy replied:
works well even though it is a hack

meji46 replied:
This is still a hack. Changing the aura type prevents the player from being invisible (blizzlike)

grimgravy replied:
I was surprised by this, I didn't imagine that the invisible effect would be impaired. However, the normal effect ( without the modification) causes skill problems. The control does not work as expected. As a result, the mission cannot be completed

Aokromes replied:
anyone can unhack this properly? this quest is needed to get horde outland quest achievment.

grimgravy replied:
@meji46 i made a recent change. You can test it again

meji46 replied:
But the hack is still there and `GetCaster()->SetVisible()` isn't a valid workaround

Aokromes replied:
sooo, anyone can fix this hack properly?

meji46 commented on src/server/scripts/Outland/zone_blades_edge_mountains.cpp:
```diff
@@ -1013,6 +1014,53 @@ class spell_oscillating_field : public SpellScriptLoader
+    void OnCharmed(bool apply) override
+    {
+        if (apply)
+            me->SetLevel(me->GetCharmerOrOwner()->GetLevel());
```
This depends on the absence of a SummonProperties flag (it should be done automatically (NYI))

meji46 commented on sql/updates/world/3.3.5/2023_xx_07_xx_world.sql:
```diff
@@ -0,0 +1,15 @@
+
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 13) AND (`SourceGroup` = 1) AND (`SourceEntry` = 38629) AND (`SourceId` = 0) AND (`ElseGroup` = 0) AND (`ConditionTypeOrReference` = 31) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 3) AND (`ConditionValue2` = 22356) AND (`ConditionValue3` = 0);
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 13) AND (`SourceGroup` = 1) AND (`SourceEntry` = 38629) AND (`SourceId` = 0) AND (`ElseGroup` = 1) AND (`ConditionTypeOrReference` = 31) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 3) AND (`ConditionValue2` = 22367) AND (`ConditionValue3` = 0);
+DELETE FROM `conditions` WHERE (`SourceTypeOrReferenceId` = 13) AND (`SourceGroup` = 1) AND (`SourceEntry` = 38629) AND (`SourceId` = 0) AND (`ElseGroup` = 2) AND (`ConditionTypeOrReference` = 31) AND (`ConditionTarget` = 0) AND (`ConditionValue1` = 3) AND (`ConditionValue2` = 22368) AND (`ConditionValue3` = 0);
```
Delete by SourceTypeOrReferenceId, SourceGroup and SourceEntry only (SourceId is always 0 for this SourceTypeOrReferenceId)

grimgravy commented:
I have a crash problem with the "void OnCharmed" output. [Link Crash](https://www.mediafire.com/file/a5v900skwabwhp7/84b8d4a1891f%252B_worldserver.exe_%255B11-4_18-19-52%255D.txt/file)

grimgravy commented:
@meji46 and @Aokromes, recheck this update I made

meji46 committed:
Core/Misc: Added new CanSeeOrDetectExtraArgs to CanSeeOrDetect (#30882)

meji46 committed:
Core/Vehicles: Added field to set a default pitch (#30878)

Naddley committed:
DB/Drustvar: Implement Quest: "Rescue Rangers"

Naddley committed:
DB/Drustvar: Implement Quest: "Total Cairnage"

Shauren committed:
Scripts/Misc: Update ScriptPCH content to include most commonly headers

Naddley committed:
DB/Drustvar: Added phased spawns after completion of quest "Rescued Ranger"

aquadeus committed:
Scripts/Spells: Fix proc of hunter talent Bullseye (#30788)

Shauren committed:
Core/PacketIO: Improved bit functions - no longer reads each bit separately

jasongdove committed:
Core/Network: Update LoginRESTService to check all addresses together (#30927)

Shauren committed:
Core/Misc: Make string and string_view literal suffixes globally available

TheSCREWEDSoftware committed:
DB/KhazAlgar: Added remaining Flightmaster spawns

- Isle of Dorn, Ringing Deeps, Hallowfall, Azj-Kahet

Closes #30940

Co-authored-by: Naddley <64811442+Naddley@users.noreply.github.com>

Shauren committed:
Core: Updated allowed build to 11.1.5.60822

Naddley committed:
Scripts/Drustvar: Implement Quest: "Really Big Problem" (#30939)

Naddley committed:
DB/Boralus: Added missing questender for Quest: "Drustvar"

Shauren committed:
Core/PacketIO: Fixed WorldPackets::SizedCString read function (it should not attempt to read null terminator if empty)

Shauren committed:
Core/PacketIO: Rename ByteBuffer::contents to ByteBuffer::data

Shauren committed:
Core/Conditions: Catch exceptions caused by invalid WorldStateExpression data

Shauren committed:
Core/Conditions: Fix copypaste from previous commit

meji46 committed:
Scripts/TheJadeForest: Implement quest "Paint it Red!" (31765) (#30884)

mdX7 committed:
DB/Templates: Added 11.1.5 wdb data

mdX7 committed:
DB/Locales: Added 11.1.5 locales

mdX7 committed:
DB/Templates: Added more 11.1.5 wdb data

mdX7 committed:
DB/Quest: Added 11.1.5 quest poi data

Shauren committed:
Core/PacketIO: Move ObjectGuid and Position::PackedXYZ packet functions out of ByteBuffer

Shauren committed:
Core/Spells: Remove NYI labels from implemented spell effect attributes

meji46 committed:
Core/SpawnTracking: Check if the objective is completable to get correct spawn tracking state (#30923)

Shauren committed:
Core/PacketIO: Migration to PacketUtilities bit/string operations part 1

Naddley committed:
DB/Drustvar: Added some more spawns

Naddley committed:
DB/Drustvar: Handle Krazzlefrazz and Swiftwind Outpost phases

+ clientside gameobject spawns

Shauren committed:
Core/Spells: Expose spell channel duration to scripts

Shauren committed:
Core/Auras: Initialize aura spell visual from spell that created it instead of recalculating it

Shauren committed:
Core/PacketIO: Implemented CMSG_UPDATE_SPELL_VISUAL

aquadeus committed:
Scripts/Spells: Implement demon hunter talent "Demonic" (#30544)

aquadeus committed:
Scripts/Spells: Implement death knight talent Sanguine Ground (#30955)

Shauren committed:
Core/PacketIO: Added missing packet util for non-bits container sizes

Shauren committed:
Core/Commands: Fixed parsing enum chat command arguments when they have a TITLE comment

Shauren committed:
Core/Commands: Don't validate enum numeric values for flags

Shauren committed:
Core/Commands: Modernize string parsing for EXACT_SEQUENCE command arguments

Shauren committed:
Core/Misc: Silence some statis analysis false positive warnings

Shauren committed:
Core/Position: Remove Position implicit object slicing

Shauren committed:
Core/Misc: Fixed some uninitialized variables

Shauren committed:
Core/Scripts: Fixed WorldMapScript being unintentionally restricted to one script for a given map id

Shauren committed:
Core/Misc: Add support for non-copyable types in LockedQueue

aquadeus committed:
Scripts/Spells: Implement death knight talent Blooddrinker (#30956)

aquadeus committed:
Scripts/Spells: Implement death knight talent Voracious (#30960)

Shauren committed:
Scripts/Spells: Fixed Improved Death Strike value calculation for blood specialization

aquadeus committed:
Scripts/Spells: Implement death knight talent Hemostasis (#30962)

aquadeus committed:
Scripts/Spells: Fix proc of death knigth talent Coagulopathy (#30961)

TheSCREWEDSoftware committed:
DB/Tele: Added more teleport points

Closes #30942

Naddley committed:
DB/Dornogal: Added Flames Radiance Recruiter spawn

Shauren committed:
Core/PacketIO: Add char overloads to ByteBuffer operators

Shauren committed:
Core/PacketIO: Allow disabling utf8 validation in SizedString/SizedCString utils

Shauren committed:
Core/PacketIO: Migration to PacketUtilities bit/string operations part 2

meji46 committed:
DB/Hotfixes: Updated hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated deDE hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated esES hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated esMX hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated frFR hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated itIT hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated koKR hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated ptBR hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated ruRU hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated zhCN hotfixes to 11.1.5.60822

meji46 committed:
DB/Hotfixes: Updated zhTW hotfixes to 11.1.5.60822

meji46 committed:
DB/Templates: Update quest data

Shauren committed:
Core/PacketIO: Disable implicit bool to int conversions for ByteBuffer operators

Shauren committed:
Core/PacketIO: Fixed SMSG_UPDATE_OBJECT structure for OptionalUpdateField<WorldPackets::Item::ItemInstance> typed fields

Shauren committed:
DB/Achievements: Added reward mail data for achievement 40945

Shauren committed:
Core/PacketIO: Migration to PacketUtilities bit/string operations part 3

An-Zheng committed:
Core/Network: Fix typo in AsyncAcceptor function name (#30975)

Shauren committed:
Core/PacketIO: Support WorldPackets::String in SizedString/SizedCString writers

Shauren committed:
Core/PacketIO: Migration to PacketUtilities bit/string operations part 4

Shauren committed:
Core/PacketIO: Mark packet parsing exception throwing code paths as [[noreturn]]

Shauren committed:
Core/PacketIO: Remove ByteBuffer constructor from MessageBuffer

Shauren committed:
Core/PacketIO: Replace static_asserts with concepts and add them to PacketUtilities as well

Shauren committed:
Core/PacketIO: Move all packet writing utilities to separate file

Shauren committed:
DB/Achievements: Added reward mail data for achievement 40942 and 40943

Golrag committed:
Battlegrounds/SeethingShore: Fix crash on player logout

aquadeus committed:
Scripts/Spells: Fix warrior talent Avatar removing movement impairing auras (#30985)

aquadeus committed:
Scripts/Spells: Implement paladin talent Divine Auxiliary (#30984)

meji46 committed:
DB/SpawnTracking: Update template data

aquadeus committed:
Scripts/Spells: Implement rogue talent Deepening Shadows (#30980)

aquadeus committed:
Scripts/Spells: Implement rogue talent Shot in the dark (#30979)

aquadeus committed:
Scripts/Spells: Implement rogue talent Cloaked in Shadows (#30974)

jasongdove committed:
DB/WesternPlaguelands: Remove duplicate Scholomance doors (#30977)

jasongdove committed:
DB/EasternPlaguelands: Remove duplicate Stratholme service entrance gates (#30976)

aquadeus committed:
Scripts/Spells: Implement rogue talent Shuriken Tornado (#30972)

aquadeus committed:
Scripts/Spells: Implement demon hunter talent Violent Transformation (#30969)

Shauren committed:
Scripts/Spells: Fixed Demonic talent starting Metamorphosis cooldown for Vengeance demon hunters

aquadeus committed:
Core/Auras: Implement PROC_FLAG_2_SUCCESSFUL_DISPEL (#30953)

TheSCREWEDSoftware committed:
DB/BurningSteppes: Update Blackrock Mountain Instance enter/exit locs

Closes #30978

jasongdove committed:
DB/Dungeons: Add instance portals for TBC dungeons (#30986)

jasongdove committed:
DB/Dungeons: Added Stratholme service entrance Areatrigger

- Adds cosmetic entrance/exit portals for Shadowfang Keep, The Deadmines, Scholomance, and Lower Blackrock Spire
- Fixes Scholomance exit AT destination

Shauren committed:
CI/Circle: Switch images from ubuntu 22.04 to debian 12 (#31002)

Naddley committed:
DB/BlackRookHold: Added init. spawns

aquadeus committed:
DB/Azsuna: Added Lyndras spawn to Lyndras Exile

Closes #29969

jasongdove committed:
DB/Dungeons: Add instance portals for WotLK dungeons (#30999)

jasongdove committed:
Core/Players: Fix player outdoor status after leaving instance (#31001)

aquadeus committed:
Scripts/Spells: Implement rogue talent Night Terrors (#30994)

jasongdove committed:
DB/Dungeons: Remove deprecated The Black Morass attunement (#30997)

aquadeus committed:
Scripts/Spells: Implement death knight talent Crimson Scourge (#30966)

Naddley committed:
DB/CathedralOfEternalNight: Added init. spawns

Naddley committed:
Scripts/Stormwind: Script second part of Quest: "The Nation of Kul Tiras" (#30964)

meji46 committed:
Core/GameObjects: Added spell cast for GAMEOBJECT_TYPE_CHEST (#30996)

aquadeus committed:
Scripts/Spells: Implement paladin talent A just reward (#30954)

Shauren committed:
Core/Items: Classify wands as ranged weapons (fixes wand shooting)

aquadeus committed:
Scripts/Spells: Implement monk talent Song of Chi-ji (#30952)

Shauren committed:
Core/Movement: Spline<float> compile fix

Shauren committed:
Core/AreaTriggers: Fixed areatrigger spline duration calculation

Shauren committed:
Core/PacketIO: Fixed structure for SMSG_DISPLAY_TOAST

Shauren committed:
Core/Scripts: Allow accessing removed spell HitAura in scripts

Naddley committed:
DB/TheArcway: Added init. spawns

jasongdove committed:
Scripts/RedridgeMountains: Implement Quest "Saving Foreman Oslow" (#30947)

Shauren committed:
Core: Updated allowed build to 11.1.5.61122

aquadeus committed:
Scripts/Spells: Fix warlock talent Shadowburn (#30951)

jasongdove committed:
Scripts/AbyssalDepths: Add whirlpool entrance to Abyssal Maw (#31007)

aquadeus committed:
Scripts/Spells: Implement warlock talent Soul Fire (#30950)

Naddley committed:
Scripts/Boralus: Implement Quest: "Daughter of the Sea" (#30967)

Commit:
TDB 1115.25051 - 2025/05/31

aquadeus committed:
Scripts/Spells: Implement hunter talent Streamline (#30941)

Shauren committed:
Core/Spells: Don't set TARGET_FLAG_DEST_LOCATION for cone targets

jasongdove committed:
DB/Dungeons: Add instance portals for Cata dungeons (#31018)

jasongdove committed:
DB/Dungeons: Add instance portals for MoP dungeons (#31019)

Shauren committed:
Core/Misc: Move fmt::formatter forward declaration to a common header to reduce number of replacements needed when upgrading fmt to new major version

Shauren committed:
Core/Misc: Add end iterator/sentinel to make `char const*` work with std::ranges algorithms

Shauren committed:
Core/Network: Make ip address formattable with fmt

Naddley committed:
DB/DalaranLegion: Script clone event for Emissary Auldbridge

Golrag committed:
Scripts/Spells: Fix Shockwave and implement Rumbling Earth talent (#31020)

Shauren committed:
Core: Updated allowed build to 11.1.5.61188

Shauren committed:
Core/Spells: Replace MAX_SPELL_EFFECTS loop limits with correct upper bound depending on how many effects the spell has v2

jasongdove committed:
DB/Dungeons: Add instance portals for WoD dungeons (#31022)

+ Added serverside exit areatrigger for The Everbloom

Shauren committed:
Core/PacketIO: Remove duplicated areatrigger spline packet writer

Naddley committed:
DB/CourtOfStars: Added init. spawns

Shauren committed:
Core: Updated allowed build to 11.1.5.61265

Shauren committed:
Core/PacketIO: Implemented CMSG_UPDATE_AREA_TRIGGER_VISUAL

Shauren committed:
Core/Chat: Remove fmt::sprintf from headers

Shauren committed:
DB/RBAC: Add missing command permission assignments and remove unused permission

Golrag committed:
Scripts/Spells: Fix warrior charge (#31017)

Shauren committed:
Scripts/Spells: Improved Glyph of the Blazing Trail visual

Shauren committed:
Core/Spells: Fixed applying glyphs without exclusive category

Shauren committed:
Core/Grids: Fix TypeListContainer::Size

Shauren committed:
Core/Grids: Remove unused template parameter from Grid class

aquadeus committed:
Scripts/Spells: Fix target limit of monk talents Revival/Restoral (#31032)

aquadeus committed:
Scripts/Spells: Implement monk talent Burst of Life (#31031)

jasongdove committed:
DB/Dungeons: Add instance portals for Legion dungeons (#31033)

+ Added serverside Areatrigger for enter/exit Legion dungeons

aquadeus committed:
Scripts/Spells: Implement monk talent Mists of Life (#31030)

Golrag committed:
Script/Spells: Fix warrior mortal strike (#31029)

Golrag committed:
Script/Spells: Remove outdated Sudden Death spell script (#31028)

Naddley committed:
DB/DarkheartThicket: Added init. spawns

Shauren committed:
Tools/Extractors: Skip extracting visual only liquids (determined by LiquidMaterial flags)

Golrag committed:
Scripts/Spells: Implement Invigorating Fury talent (#31040)

Golrag committed:
DB/Proc: Add Hack and Slash proc data (#31042)

Shauren committed:
Core/Movement: Fixed orientation calculation when using MoveSpline::ComputePosition(int32 time_offset) overload

Golrag committed:
Scripts/Spells: Make Rampage trigger enrage (#31041)

Shauren committed:
Fix sql name

Shauren committed:
Core/Misc: Kill AnyPlayerInObjectRangeCheck (duplicate of AnyUnitInObjectRangeCheck)

Shauren committed:
Core/Misc: Remove boost/algorithm dependency

Golrag committed:
Scripts/Spells: Implement Frenzied & Powerful Enrage talents (#31039)

Golrag committed:
Scripts/Spells: Implement Raging Blow cooldown reset talents (#31038)

Golrag committed:
Scripts/Spells: Correctly handle enrage on rampage & bloodthirst (#31045)

Shauren committed:
Core/Spells: Allow removing weapon damage effects by preventing target selection in scripts

Naddley committed:
DB/RagefireChasm: Added loot for Boss encounters

Closes #30666

Naddley committed:
DB/RagefireChasm: Added loot for trash mobs

Closes #30667

Naddley committed:
DB/VaultOfTheWardens: Added init. spawns

Todo: movement for Viletongue Belcher and Malignant Defiler

Golrag committed:
Scripts/Spells: Fix Odyn's Fury target cap & Titanic Rage talent (#31043)

Golrag committed:
DB/Spells: Add proc data for Bloodcraze (#31051)

Shauren committed:
Core/Misc: Kill copy/move constructors for singleton classes

meji46 committed:
DB/TirisfalGlades: Fix talk targets for quest "The Shadow Grave"

Closes #31058

Shauren committed:
Core/Movement: Fix WaypointMgr::GetNode infinite recursion (function isn't used anywhere currently)

Naddley committed:
DB/TirisfalGlades: Update quest "The Wakening" and delete old scripts

Closes #31061

Shauren committed:
Core/Misc: Fix revision_data.h and boost/predef preprocessor macro conflict

Shauren committed:
Core/Common: Update PCH content to include most commonly used headers

Shauren committed:
Core/Database: Update PCH content to include most commonly used headers

Shauren committed:
Dep/Protobuf: Enable PCH

Shauren committed:
Dep/g3d: Enable PCH

Shauren committed:
Core/Network: Enable PCH

Shauren committed:
Core/Proto: Update PCH content to include most commonly used headers

aquadeus committed:
Scripts/Spells: Implement monk talent Keg Smash target limit (#31048)

aquadeus committed:
Scripts/Spells: Implement hunter talent Precise Shots (#31049)

Shauren committed:
Core/Shared: Update PCH content to include most commonly used headers

Shauren committed:
Core/Bnet: Update PCH content to include most commonly used headers

Shauren committed:
Core/Game: Update PCH content to include most commonly used headers

Shauren committed:
Core/Worldserver: Update PCH content to include most commonly used headers

Shauren committed:
Core/Scripts: Update PCH content to include most commonly used headers

Shauren committed:
Core/Misc: Remove a bunch of code from headers

Shauren committed:
Core/Scripts: Mark script base class constructors noexcept

Shauren committed:
Core/Players: Fixed traits not updating aura values when changing rank

aquadeus committed:
Scripts/TheStonevault: Implement Skarmorak encounter (#30937)

Naddley committed:
Scripts/DalaranLegion: Implement aura to handle Orderhall Campaign Intro (#31063)

Naddley committed:
DB/SeatOfTheTriumvirate: Added init. spawns

aquadeus committed:
Scripts/WanderingIsle: Implement quest 29420 (The Spirit's Guardian) (#30467)

Golrag committed:
Scripts/Spells: Fix warrior bloodthirst heal (#31027)

Golrag committed:
Scripts/Spells: Implement Frenzy fury warrior talent (#31075)

Golrag committed:
Scripts/Spells: Implement Deft Experience warrior talent (#31060)

Golrag committed:
Scripts/Spells: Implement Vicious Contempt warrior talent (#31055)

Golrag committed:
Scripts/Spells: Implemented Menace warrior talent (#31015)

aquadeus committed:
Scripts/Spells: Implement druid talent Lunar Beam (#30777)

jasongdove committed:
Core/AreaTriggers: Add AREATRIGGER_ACTION_TAVERN (#31035)

Naddley committed:
DB/Stormwind: Some updates to Valley of Heroes

Naddley committed:
DB/DalaranLegion: Implement Quest: "Felstorms Plea"

Naddley committed:
DB/DalaranLegion: Implement Quest: "A Summons From Moonglade"

Shauren committed:
DB/Achievements: Added reward mail data for achievement 40944

Naddley committed:
DB/Spell: Added spelltarget pos. data for Teleport: Moonglade

Naddley committed:
DB/DalaranLegion: Script Initiate Da-Nel for Quest: "Before the Storm"

jasongdove committed:
DB/SiegeOfBoralus: Show exit portals and ATs in normal difficulty (#31078)

Shauren committed:
Core/Calendar: MSVC nopch fix

Naddley committed:
DB/DalaranLegion: Implement Quest: "Needs of the Hunters"

Shauren committed:
Core/Misc: Remove extremely forgettable tiny utility function

Shauren committed:
Core/Objects: Replace separate x,y,z,o getters for stationary position with single Position

Shauren committed:
Core/Movement: Spline code cleanup

* Remove unneccessary reserve in MoveSplineInitArgs constructor - MoveTo always resizes it
* Remove unused function computeDuration
* Split cyclic spline reinitialization to separate function
* Don't expose TransportPathTransform outside of MoveSplineInit
* Ensure function argument names are the same in declaration and definition

Shauren committed:
Core: Updated to 11.1.7

Shauren committed:
Tools/Extractors: Update db2 metadata

Naddley committed:
DB/DalaranLegion: Implement Quest: "The Hunters Call"

Shauren committed:
Core: Updated allowed build to 11.1.7.61559

Naddley committed:
DB/EyeOfAzshara: Added init. spawns

Shauren committed:
Core/Conditions: Implemented new condition source type CONDITION_SOURCE_TYPE_PLAYER_CHOICE_RESPONSE

Shauren committed:
Core/Database: Fixed DEFINE_FIELD_ACCESSOR_CACHE_ANONYMOUS macro arguments

Shauren committed:
Core/Gossip: Use algorithms from std::ranges namespace in gossip menu code

Shauren committed:
Core/Gossip: Use std::variant to store InteractionData

Naddley committed:
DB/NeltharionsLair: Added init. spawns

Shauren committed:
Core/Conditions: Fix validation for CONDITION_SOURCE_TYPE_PLAYER_CHOICE_RESPONSE

Shauren committed:
Core/Players: PlayerChoice improvements

* Add missing choice properties to database (InfiniteRange, ShowChoicesAsList)
* Allow limiiting the number of responses sent at the same time
* Fixed duration sent in SMSG_DISPLAY_PLAYER_CHOICE
* Remove dynamically generated response identifiers from database
* Remove auto rewarding choice responses
* Change response scripts to be bound to scriptname

Shauren committed:
SQL: Fix missing comma

Naddley committed:
DB/Karazhan: Added init. spawns for lower return to Karazhan

Shauren committed:
Core/Players: Add missing playerchoice expiration initialization

Shauren committed:
Core/Misc: Remove StringFormat exception handling code from the header

Shauren committed:
Core/Database: Add new member functions to Field class that return empty optionals if database result was NULL

Shauren committed:
Core/Misc: Use new Field::Get*OrNull where possible

grimgravy commented:
Could someone test this PR?

Shauren committed:
Core/Commands: Enable setting linked aggro and formation movement with .npc add formation and fix swapped saved dist/angle values

Shauren committed:
Core: Updated allowed build to 11.1.7.61609

Shauren committed:
Core/Commands: Waypoint command fixes

* .wp add will now add data to waypoint_path table
* .wp reload will no longer crash the server
* Replace deprecated command handler arguments

Shauren committed:
Core/Database: Improve code generated by DEFINE_FIELD_ACCESSOR_CACHE

Shauren committed:
DB/Achievements: Add mail rewards for achievements 40882 and 41211

Shauren committed:
Core/LFG Fix crash on lfg boss kills

Shauren committed:
Core/LFG: Change InstanceScript::UpdateLfgEncounterState to only pass valid dungeon encounter ids to LFGMgr

Shauren committed:
Core/Chat: Fixed chat language translations for messages containing item links using new 11.1.5 color format

meji46 committed:
DB/Hotfixes: Updated hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated deDE hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated esES hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated esMX hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated frFR hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated itIT hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated koKR hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated ptBR hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated ruRU hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated zhCN hotfixes to 11.1.7.61609

meji46 committed:
DB/Hotfixes: Updated zhTW hotfixes to 11.1.7.61609

Naddley committed:
DB/Karazhan: Added init. spawns for upper return to Karazhan

grimgravy committed:
update

grimgravy committed:
update
